core/control-plane/control-plane-aggregate-services/src/test/java/org/eclipse/edc/connector/service/policydefinition/PolicyDefinitionServiceImplTest.java

```diff
@@ -90,6 +90,18 @@ void search_invalidExpression_raiseException(Criterion invalidFilter) {
         assertThat(result).isFailed();
     }
 
+    @ParameterizedTest
+    @ArgumentsSource(ValidFilters.class)
+    void search_validExpression_privateProperties(Criterion validFilter) {
```

Comment:
I'm not sure about the value added by this test, this behavior should be tested on the validator and eventually it could be injected and mocked for testing what happens when the validation fails

Reply:
This test ensures what kind of jsonpath allowed

Reply:
This test should be moved on the `QueryValidator` test. 
In any case, my suggestion is to have a single "working" test and then tests that shows which kind of inputs are not accepted. This is a common pattern in testing validators

Reply:
EDIT: I saw that currently they are tested this way, in my opinion this should be changed as not optimal (testing a validator passing through the service is in fact making this an integration test), but it is ok for the moment to keep them this way, I opened a PR to address the refactoring: https://github.com/eclipse-edc/Connector/issues/3731